src/Prima/Pyxis/Form/Autocomplete.elm

```diff
@@ -92,51 +111,161 @@ The configuration of the `Autocomplete`'s `State`.
 type alias StateConfig =
     { focused : Maybe String
     , selected : Maybe String
-    , filter : Maybe String
+    , filter : Filter
     , isMenuOpen : Bool
-    , choices : List AutocompleteChoice
+    , choices : ChoicesStatus
+    , quiteFor100ms : Debouncer.Debouncer Msg Msg
+    , threshold : Int
     }
 
 
 {-| Initializes the `Autocomplete`'s `State`.
 -}
-init : List AutocompleteChoice -> State
-init choices =
-    State <| StateConfig Nothing Nothing Nothing False choices
+init : State
+init =
+    State <| StateConfig Nothing Nothing Nothing False Pristine (initDebouncer 0) 1
+
+
+{-| Internal. Debouncer initializer
+-}
+initDebouncer : Float -> Debouncer.Debouncer o o
+initDebouncer secondsDebounce =
+    Debouncer.manual
+        |> Debouncer.settleWhenQuietFor (Just <| Debouncer.fromSeconds secondsDebounce)
+        |> Debouncer.toDebouncer
 
 
 {-| Updates the `Autocomplete`'s `State`.
 -}
-update : Msg -> State -> State
+update : Msg -> State -> ( State, Cmd Msg, Filter )
 update msg ((State state) as stateModel) =
     case msg of
-        OnFilter value ->
-            updateOnFilter (Just value) stateModel
+        OnInput value ->
+            stateModel
+                |> updateOnInput (Just value)
+                |> H.withCmds [ send <| Debounce (Debouncer.provideInput OnFilter) ]
+                |> withoutFilter
+
+        Debounce subMsg ->
```

Comment:
Per quanto possibile questa logica dovrebbe essere splittata in varie funzioni. 
L'update dovrebbe avere meno `let ... in ` possibile e rimanere chiara e imperativa

Reply:
però in questo caso è molto chiaro e semplice, non esageriamo con sta ossessione per i pipe....